Rubberduck.VBEEditor/ComManagement/TypeLibs/UnmanagedMemHelper.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Rubberduck.VBEditor.ComManagement.TypeLibs
+{
+    /// <summary>
+    /// Windows API constants used by VirtualQuery, see https://msdn.microsoft.com/en-us/library/windows/desktop/aa366786.aspx
+    /// </summary>
+    public enum ALLOCATION_PROTECTION : uint
+    {
+        PAGE_EXECUTE = 0x00000010,
+        PAGE_EXECUTE_READ = 0x00000020,
+        PAGE_EXECUTE_READWRITE = 0x00000040,
+        PAGE_EXECUTE_WRITECOPY = 0x00000080,
+        PAGE_NOACCESS = 0x00000001,
+        PAGE_READONLY = 0x00000002,
+        PAGE_READWRITE = 0x00000004,
+        PAGE_WRITECOPY = 0x00000008,
+        PAGE_GUARD = 0x00000100,
+        PAGE_NOCACHE = 0x00000200,
+        PAGE_WRITECOMBINE = 0x00000400
+    }
+
+    /// <summary>
+    /// Windows API structure used by VirtualQuery, see https://msdn.microsoft.com/en-us/library/windows/desktop/aa366775.aspx
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential)]
+    public struct MEMORY_BASIC_INFORMATION
+    {
+        public IntPtr BaseAddress;
+        public IntPtr AllocationBase;
+        public uint AllocationProtect;
+        public IntPtr RegionSize;
+        public uint State;
+        public ALLOCATION_PROTECTION Protect;
+        public uint Type;
+    }
+
+    /// <summary>
+    /// Exposes some special routines for dealing with unmanaged memory
+    /// </summary>
+    public static class UnmanagedMemHelper
```

Comment:
Nit: I'd name it `UnmanagedMemoryHelper` -- `Mem` could be read as `member` or `memory`.

Reply:
agreed, will sort.